Player Input & Movement improvements.

ValgardurRI opened the pull request:
changed the mesh generateor to also generate a mesh collider
changed all aspects of movement to be on the xz axis
added 3d colliders to prefabs for bullet and player

NOTE: started working on the navMeshExample because i'm forgetful.
If anyone else has been editing it, they can discard my version.

ValgardurRI committed:
added pathfinding solution to the project, needed to place it in its own namespace due to naming conflicts

ValgardurRI committed:
Merge branch 'master' of https://github.com/video-game/final-game into FuckYouUnity-HowAboutSome2DNavMesh

ValgardurRI committed:
fixed player movement to work in the xz axis

ValgardurRI committed:
fixed projectiles not colliding with the environment

ValgardurRI committed:
Merge branch 'master' of https://github.com/video-game/final-game into fixplayermovement

ValgardurRI committed:
merged with master to fix conflicts

dadisigursveinn commented on Assets/Prefabs/Player.prefab:
```diff
@@ -54,87 +53,42 @@ Transform:
   m_GameObject: {fileID: 1630354825624260}
-  m_LocalRotation: {x: 0, y: 0, z: 0, w: 1}
-  m_LocalPosition: {x: 0, y: 0, z: 0}
+  m_LocalRotation: {x: 0.7071068, y: 0, z: 0, w: 0.7071068}
```
heh

dadisigursveinn commented on Assets/Scripts/DemoProjectileScript.cs:
```diff
@@ -13,7 +13,7 @@ private void Awake()
 
-	private void OnCollisionEnter2D(Collision2D other)
+	private void OnCollisionEnter(Collision other)
     {   
```
good indent

dadisigursveinn commented on Assets/Scripts/TileMapToMesh.cs:
```diff
@@ -17,7 +17,7 @@ public class TileMapToMesh : MonoBehaviour {
     GameObject meshContainer;
     const string containerName = "Mesh Container";
-
+    
```
https://marketplace.visualstudio.com/items?itemName=SteveCadwallader.CodeMaid

dadisigursveinn reviewed (changes requested):
Builds folder, why.

bjornt16 committed:
removed builds folder .. how'd this even get in here.

bjornt16 committed:
removed Post-fix "Script" from Player and DemoProjectile

bjornt16 committed:
Moved Direction checking to Utilities.

bjornt16 committed:
updated DemoProjectile prefab to match changes

bjornt16 committed:
GameManager now keeps track of multiple players (just incase)

bjornt16 committed:
added a sorting layer for projectiles

bjornt16 committed:
Refactored Player

Player no longer uses hardcoded input handling.
Input handling moved to InputManager.
Movement & dash is now done with NavMesh (for consistency and now we only need mesh colliders on walls for projectiles)

bjornt16 committed:
InputManager now does the input handling.

Improved PlayerControl class to easily check for player movement / aim.

bjornt16 committed:
Updated Player Prefab to match changes.

bjornt16 committed:
Saved changes to scene

bjornt16 committed:
removed TileMapToMesh changes to avoid merge conflicts with pull request #21

I didnt see these changes before i started updating TileMapToMesh.
I have also added Mesh collider generation, as well as some other QoL improvements.

bjornt16 committed:
added some comments

bjornt16 commented:
I moved Player Input handling to Input Manager.
Movement now uses the NavMesh, so we won't have to create walls with mesh colliders everywhere. (except for projectiles .. players cant move on water, but projectile should go over water .. so it works out nicely). This also has the added benefit of the player not getting stuck on walls.

I also removed @ValgardurRI TileMapToMesh changes .. I had already finished a bunch of improvements to the script before I noticed that this pull request had changes to the script. (including adding mesh colliders on generate)  .. so hopefully this will reduce any conflicts with pull request #21

hallur commented on ProjectSettings/ProjectVersion.txt:
```diff
@@ -1 +1 @@
-m_EditorVersion: 2017.3.1f1
+m_EditorVersion: 2017.3.0f3
```
Hmm

bjornt16 replied:
This will happen everytime someone with a different version opens the project... it's to be expected .. as long as no one is using an ancient version it's fine.

hallur commented on Assets/Scripts/Player.cs:
```diff
@@ -0,0 +1,101 @@
+    public void Move(float horizontal, float vertical)
+    {
+        isWalking = (horizontal != 0 || vertical != 0);
+        animator.SetBool("IsWalking", (horizontal != 0 || vertical != 0));
```
Unused variable isWalking. Should probably just remove it

bjornt16 replied:
whoops. I removed it, then put it back, forgot to add it back to the SetBool function.. might be useful later.

hallur commented on Assets/Scripts/Managers/GameManager.cs:
```diff
@@ -60,10 +59,10 @@ public void EndGame()
+        for (int i = 0; i < p.Length; i++)
         {
-            player = p.gameObject;
+            player.Add(p[i]);
```
This can be done in a single line "player.AddRange(GameObject.FindObjectsOfType<Player>());"

hallur commented on Assets/Scripts/InputManager.cs:
```diff
@@ -52,8 +59,80 @@ private void Update()
+        get
+        {
+            return (Input.GetKeyUp(MoveUp) || Input.GetKeyUp(MoveDown) ||
+                   Input.GetKeyUp(MoveLeft) || Input.GetKeyUp(MoveRight));
```
Won't this incorrectly return true when, for example, the player is holding down MoveUp and MoveRight, and then when he releases MoveRight but is still holding MoveUp?

bjornt16 replied:
Yes, but it doesn't matter, I pass a move value to the player.. and it's based off what buttons are down. The GetKeyUp stuff is just so i can call move again .. but with move values of 0.

hallur commented on Assets/Scenes/navmesh fixedplayer.unity:
```diff
@@ -0,0 +1,9758 @@
+%YAML 1.1
```
Is this scene supposed to be included?

bjornt16 replied:
ill delete it @ValgardurRI  is done with it.

hallur commented:
Player projectiles spawn inside player instead of at his hand

Dash length does not increase when dashDuration is increased

bjornt16 committed:
nitpickery

bjornt16 committed:
created projectile spawn point

bjornt16 committed:
removed unused scene

bjornt16 commented:
fixed

bjornt16 committed:
changed projectileSpawn to Transform instead of vector3
